RealmSwift/Tests/CombineTests.swift

```diff
@@ -842,14 +842,16 @@ private class CombineCollectionPublisherTests<Collection: RealmCollection>: Comb
 
     func testChangeSetWithoutNotifying() {
         var calls = 0
+        let ex = expectation(description: "sink init")
         cancellable = collection
             .changesetPublisher
             .saveToken(on: self, at: \.notificationToken)
             .sink { _ in
                 calls += 1
+                ex.fulfill()
             }
         XCTAssertNotNil(notificationToken)
-
+        waitForExpectations(timeout: 2.0, handler: nil)
```

Comment:
This wait shouldn't be needed and if it does fix the test then it's just working around a bug which the test was exposing.

Reply:
The test always passes for me locally, this was to test on the CI.

Why isn't this wait needed though? Isn't the initial `sink` value still called asynchronously?

Reply:
I think maybe rather than this wait being unnecessary, there's actually two different things that should be tested. Adding a notifier and then immediately skipping it before receiving the initial notification is something that should work. This is what the test was previously testing on CI, but for that it needs a wait *after* the writes because it's possible for all of them to happen before initial is ever delivered. Adding the wait here validates that it successfully skips after receiving the initial notification, which of course it also something that should be tested.

So basically the test is correct, but there should be another copy of this test added with the wait after the loop but before the assert on call count.